### PR TITLE
[editorial] Unify built-in function tables

### PR DESCRIPTION
**Pull request by alan-baker**
Fix missing descriptions and unify styles of tables.

### CODE REVIEW

**Review by kvark — APPROVED** [review]
How do you feel about taking this opportunity and combining the scalar + vector variants closer to each other?
Take `abs` case for example, the "Precondition" could be a merged field between the two rows, showing `T is f32 `. Then the "Conclusion" would have 2 separate rows.
Finally, the "Description" would be merged.

**Comment by alan-baker** [comment]
> How do you feel about taking this opportunity and combining the scalar + vector variants closer to each other?
> Take `abs` case for example, the "Precondition" could be a merged field between the two rows, showing `T is f32 `. Then the "Conclusion" would have 2 separate rows.
> Finally, the "Description" would be merged.

I thought about something like that. So turn abs into:
>Returns the absolute value of |e| (e.g. |e| with a positive sign bit). Results are computed component-wise for vector types. (GLSLstd450FAbs)

The change being mentioning component-wise computation. Thoughts?

File: wgsl/index.bs [hunk 1]
```diff
@@ -6349,244 +6362,340 @@ That's not a full user-defined function declaration.
...
   <tr algorithm="scalar case, cosh">
     <td>|T| is f32
     <td class="nowrap">`cosh(`|e|`:` |T| `) -> ` |T|
-    <td>(GLSLstd450Cosh)
+    <td>Returns the hyperbolic cosine of |e|. (GLSLstd450Cosh)
   <tr algorithm="vector case, cosh">
     <td>|T| is f32
     <td class="nowrap">`cosh(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
-    <td>(GLSLstd450Cosh)
+    <td>Component hyperbolic cosine.
```
Comment (dneto0) [inline wgsl/index.bs:6463]: nit: Component-wise hyperbolic cosine.

File: wgsl/index.bs [hunk 2]
```diff
@@ -6349,244 +6362,340 @@ That's not a full user-defined function declaration.
...
-    <td>(GLSLstd450Log2)
+    <td>Component-wise base-2 logarithm.
+    Component |i| of the result is `log2(`|e|`[i])`.
+    (GLSLstd450Log2)
+
   <tr algorithm="scalar case, max">
     <td>|T| is f32
     <td class="nowrap">`max(`|e1|`:` |T| `, `|e2|`:` |T| `) -> ` |T|
-    <td>(GLSLstd450NMax)
+    <td>Returns |e2| if |e1| is less than |e2|, and |e1| otherwise. (GLSLstd450NMax)
```
Comment (dneto0) [inline wgsl/index.bs:6624]: Something somewhere should spell out the behaviour on NaNs.

I filed https://github.com/gpuweb/gpuweb/issues/1711 for followup.

File: wgsl/index.bs [hunk 3]
```diff
@@ -6349,244 +6362,340 @@ That's not a full user-defined function declaration.
...
     <td class="nowrap">`reflect(`|e1|`:` |T| `, `|e2|`:` |T| `) -> ` |T|
-    <td>(GLSLstd450Reflect)
+    <td>For the incident vector |e1| and surface orientation |e2|, returns the reflection direction
+    |e1|`-2*dot(`|e2|`,`|e1|`)*|e2|`.
+    (GLSLstd450Reflect)
   <tr algorithm="vector case, reflect">
     <td>|T| is f32
     <td class="nowrap">`reflect(`|e1|`:` vec|N|<|T|> `, `|e2|`:` vec|N|<|T|>`) -> ` vec|N|<|T|>
-    <td>(GLSLstd450Reflect)
+    <td>Component-wise relfection direction.
```
Comment (dneto0) [inline wgsl/index.bs:6695]: Spelling.  "reflection"

**Review by dneto0 — CHANGES_REQUESTED** [review]
Just a couple of typos to fix.

File: wgsl/index.bs (hunk 2 above)
Comment (alan-baker) [reply wgsl/index.bs:6624]: Agreed it should be spelled out and that it should be done separately.

**Review by dneto0 — APPROVED** [review]

**Comment by dneto0** [comment]
> > How do you feel about taking this opportunity and combining the scalar + vector variants closer to each other?
> > Take `abs` case for example, the "Precondition" could be a merged field between the two rows, showing `T is f32 `. Then the "Conclusion" would have 2 separate rows.
> > Finally, the "Description" would be merged.
> 
> I thought about something like that. So turn abs into:
> 
> > Returns the absolute value of |e| (e.g. |e| with a positive sign bit). Results are computed component-wise for vector types. (GLSLstd450FAbs)
> 
> The change being mentioning component-wise computation. Thoughts?

@dj2 and I discussed whether "component-wise" was clear enough.  It might not be familiar to all readers.  So we went with more words that were more assuredly clear.   If you want to pursue this, let's do it in another PR.

**Pull request merged**